### DISCUSSION TITLE
Error: pg_config executable not found.

### DISCUSSION THREAD

**Discussion by abdounasser202** [sbidoul/ocadays2024-odoo-project-demo#5] [Q&A] [answered]
Hello @sbidoul , nice talk and demo

Most of the time, i'm working with odoo from docker. 

With uv, i would like to work without docker and i was checking how i can run everything out of the box without installing anything in my physical machine.

So, yesterday i followed your talk and did the installation. After that, i got an error related to psql binary, so i did `uv add psycopg2-binary==2.9.9`

Then when i launch, it says that pg_config executable is not found. I understand that we may install psql in the physical machine.

But i'm wondering how i can install everything out of the box, psql included in my physical machine and without using docker

**Comment by sbidoul** [comment, marked as answer]
Yes the problem is that odoo depends on `psycopg2`, and `psycopg2-binary` is another distribution name, so uv is still trying to build and install `psycopg2` to satisfy the dependency.

So you'll need to install the C compiler and the dev dependencies. On debian/ubuntu it's something like `apt install clang build-essential libpq-dev`. Also maybe `python3-dev` depending on how you installed python.

**Reply by abdounasser202** [reply]
`apt install build-essential clang libpq-dev python3-dev ` is OK
